### ISSUE TITLE
Unable to choose XFS as root partition in clr-installer nor clr-installer-gui

### ISSUE THREAD

**Issue by jiblime** [clearlinux/clr-installer#507]
**Describe the bug**
clr-installer-gui does not have an option (that I am aware of) for choosing root partition fs. 

clr-installer, upon selecting advanced configuration, does not allow me to select more than two partitions to edit. I need three partitions to edit, being the boot, root, and swap, to confirm.

It also wiped my entire drive upon failing when attempting to use the newly made .yaml.

**To Reproduce**
1. Boot a baremetal "Live Desktop" ISO 
2. Terminal -> clr-installer
3. Go to installation media, advanced configuration
4. Try to choose more than two partitions with an encrypted XFS root, using an existing FAT32 boot. You cannot because after selecting two partitions you are no longer allowed to edit it. You can't confirm the change because the installer requires root/boot/swap.
--
5. Use safe installation, the automatic one, so a .yaml can be created. Confirm.
5. Go to advanced configuration by pressing A and create a .yaml file to use. 
6. Exit, and edit the .yaml file by changing the root partition's fs from ext4 to xfs.
7. Run clr-installer-gui -c clr-installer.yaml

**Expected behavior**
I expected to be able to finish advanced configuration for my disk, and I also expected to be prompted before having my hard drive wiped.

**Environment (please complete the following information):**
 - Clear Linux OS Version: 30900
 - Installer Version 2.1.1
 - Platform: CPU family/model: 6/94

**Additional context**

I ran `clr-installer-gui -c clr-installer.yaml` from /home/clrlinux.

[/root/clr-installer.log](https://github.com/clearlinux/clr-installer/files/3598367/clr-installer.log)
[/home/clrlinux/clr-installer.yaml](https://github.com/clearlinux/clr-installer/files/3598368/clr-installer.yaml.txt)
[root/pre-install-clr-installer.yaml](https://github.com/clearlinux/clr-installer/files/3598403/pre-install-clr-installer.yaml.txt)

When I selected my boot partition, I need to untick 'format' because it autoticks when changing the mount point.
`ls -w1` output of what used to be my EFI partition:
```
''$'\021''.'$'\023'
''$'\031''.'$'\033'
''$'\t''.'$'\v'
'!.#'
').+'
 1.3
'9.;'
 a.c
 A.C
 Q.s
'y.['
```

**Comment by mdhorn** [comment]
@jiblime Thanks for the feedback.

Using the -c option is used for non-interactive installations or image generation. 

We are in the process of creating a new release of the install which will support a new method for performing Advanced Disk configuration.  See #410 

There will also be new documentation to explain how to use this feature:
https://github.com/clearlinux/clear-linux-documentation/blob/2bc31163410c5e056fd23aabcd3e3360d5d5e820/source/get-started/bare-metal-install-desktop.rst

We are finishing some validation work, but I expect a new release of the installer by the end of the week.

**Comment by jiblime** [comment]
> @jiblime Thanks for the feedback.
> 
> Using the -c option is used for non-interactive installations or image generation.
> 
> We are in the process of creating a new release of the install which will support a new method for performing Advanced Disk configuration. See #410

Hi @mdhorn, thanks for the quick reply. That's good to hear.

In the meantime, is there a way to use XFS as root for Clear Linux, or 'continue' an installation with an edited .yaml? I don't have a sane backup of my system and planned to do an offline sync with Clear. And should I be adding suggestions to #410 or opening a new ticket (it is related to configuration)? 

I also would be happy to test the installer if that's possible. I normally use `parted`, `cryptsetup`, `mkfs.**`, and occasionally `cfdisk` to create my partitions.

**Comment by mdhorn** [comment]
> Hi @mdhorn, thanks for the quick reply. That's good to hear.
> 
> In the meantime, is there a way to use XFS as root for Clear Linux, or 'continue' an installation with an edited .yaml? I don't have a sane backup of my system and planned to do an offline sync with Clear. And should I be adding suggestions to #410 or opening a new ticket (it is related to configuration)?
> 
With the current, released installer I don't think there is a good path to do that.

> I also would be happy to test the installer if that's possible. I normally use `parted`, `cryptsetup`, `mkfs.**`, and occasionally `cfdisk` to create my partitions.
You could pull the current branch and build the new installer yourself, but building the installer requires it to be built under Clear Linux OS.
https://github.com/mdhorn/clr-installer/tree/diskUtil

You could download the live-desktop raw image
https://download.clearlinux.org/current/clear-30970-live-desktop.img.xz
It should have enough space to build the text-based version of the new installer

- Login as the default user: clrlinux
- Clone the codebase
```
git clone https://github.com/mdhorn/clr-installer.git
cd clr-installer
git checkout diskUtil
```

- Build the text-based installer and generate 
```
make build-tui
sudo .gopath/bin/clr-installer-tui -c scripts/developer-live-server.yaml --iso=false
```
- You may have to make root bigger or remove some of the desktop bundles to regain enough room.

- Exact the newly built image from the VM and place it on a USB stick
```
scp dev-clear-live-server.img <user>@<somehost>:
...
dd if=dev-clear-live-server.img of=<USB DEV> bs=4M status=progress
```


OR wait until the end of this week for the new release.

**Comment by jiblime** [comment]
Hi,

I was able to create an xfs root; I used 'auto-partition' and resized the root partition. It showed the root partition as missing after doing so, but I was able to confirm and save the .yaml and it was correct.

While on the live USB I wasn't able to install the go-basic bundle because it was too big to install. I hadn't thought of deleting other bundles beforehand. I'll give it a go after I finish restoring the rest of my data. I've checked out the newer version now that I can and I don't think issue needs to be open anymore. Thank you for your help.
